uuid/uuid.go

```diff
@@ -18,11 +18,37 @@ func NewV4() UUID {
 	return UUID(uuid.NewV4())
 }
 
-// Used in string method conversion
-const dash byte = '-'
-
-// Returns canonical string representation of UUID:
-// xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.
+// String Wrapper over the real String method
 func (u UUID) String() string {
 	return uuid.UUID(u).String()
 }
+
+// MarshalText Wrapper over the real MarshalText method
+func (u UUID) MarshalText() (text []byte, err error) {
+	return uuid.UUID(u).MarshalText()
+}
+
+// MarshalBinary Wrapper over the real MarshalBinary method
+func (u UUID) MarshalBinary() ([]byte, error) {
+	return uuid.UUID(u).MarshalBinary()
+}
+
+// UnmarshalBinary Wrapper over the real UnmarshalBinary method
+func (u *UUID) UnmarshalBinary(data []byte) error {
+	t := uuid.UUID{}
```

Comment:
It seems strange that the target is a pointer to UUID but that the code then uses a non-pointer `uuid.UUID` to perform the call, is that intended?